### PR TITLE
Fixed an added_after bug, compacted some functions, fixed a bug that …

### PR DESCRIPTION
**Pull request by zrush-mitre**
…would prevent the last parameter from being checked every time, and added lots of tests to help check for these things in the future

### CODE REVIEW

File: medallion/backends/memory_backend.py [hunk 1]
```diff
@@ -206,19 +206,20 @@ def get_object(self, api_root, collection_id, object_id, filter_args, allowed_fi
...
-                    )
-            return create_resource("objects", objs)
+                # if filter_args:
+                full_filter = BasicFilter(filter_args)
+                results.extend(full_filter.process_filter(
+                    objs,
+                    allowed_filters,
+                    collection.get("manifest", []),
+                ))
+            return create_resource("objects", results)
```
[the remark this answers is not in the document]
Comment (emmanvg) [reply medallion/backends/memory_backend.py]: (this was supposed to be from last week, I forgot to submit the review)

[the remark this answers is not in the document]
Comment (zrush-mitre) [reply medallion/backends/memory_backend.py]: That looks good, though GitHub for whatever reason is showing it as just deleting the "create_resource" line and putting the get_object function in again. I'll take your modifications and add them to the branch.

File: medallion/backends/memory_backend.py [hunk 2]
```diff
@@ -133,17 +132,14 @@ def get_objects(self, api_root, collection_id, filter_args, allowed_filters):
...
-                    else:
-                        objs.extend(collection.get("objects", []))
+                    full_filter = BasicFilter(filter_args)
+                    objs.extend(
+                        full_filter.process_filter(
+                            collection.get("objects", []),
+                            allowed_filters,
+                            collection.get("manifest", []),
+                        ),
+                    )
```
Comment (emmanvg) [inline medallion/backends/memory_backend.py]: objs.extend() is not required as we only extracting objects from a single collection.

```suggestion
                    objs = full_filter.process_filter(
                        collection.get("objects", []),
                        allowed_filters,
                        collection.get("manifest", []),
                    )
                    break
```

**Review by emmanvg — APPROVED** [review]
The PR has been aligned with latest changes. The added_after parameter functionality has been improved for MemoryBackend. Thanks!

**Pull request merged**